src/disco/components/InstallButton.js

```diff
@@ -1,16 +1,65 @@
-import React from 'react';
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
 
 import { gettext as _ } from 'core/utils';
 
 import 'disco/css/InstallButton.scss';
+import {
+  DOWNLOADING,
+  INSTALLED,
+  INSTALLING,
+  UNINSTALLED,
+  UNINSTALLING,
+  UNKNOWN,
+} from 'disco/constants';
 
 
 export default class InstallButton extends React.Component {
+
+  static propTypes = {
+    addonState: PropTypes.oneOf([
+      DOWNLOADING,
+      INSTALLED,
+      INSTALLING,
+      UNINSTALLED,
+      UNINSTALLING,
+      UNKNOWN,
+    ]),
+    downloadProgressPercent: PropTypes.number,
+    handleClick: PropTypes.func,
+    handleChange: PropTypes.func,
+  }
+
+  static defaultProps = {
+    addonState: UNKNOWN,
+    downloadProgressPercent: 0,
+  }
+
   render() {
+    const { addonState, downloadProgressPercent } = this.props;
+    const installed = addonState === INSTALLED &&
+                      addonState !== UNINSTALLED;
+    const isDisabled = addonState === UNKNOWN;
+    const isDownloading = addonState === DOWNLOADING;
+
+    const switchClasses = classNames('switch', {
+      downloading: isDownloading,
+      installed,
+      installing: addonState === INSTALLING,
+      uninstalling: addonState === UNINSTALLING,
```

Comment:
Yeah or just this would work. I don't have a class for unknown but that doesn't really matter.  

```
const switchClasses =  `switch ${addonState}`;
```